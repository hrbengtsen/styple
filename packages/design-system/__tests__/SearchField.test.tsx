import * as React from "react";
import { render, RenderResult, screen } from "@testing-library/react";
import { SearchField } from "..";

describe("Test SearchField", () => {
  let view: RenderResult;

  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, "useState");

  // @ts-ignore
  useStateSpy.mockImplementation((init) => [init, setState]);

  const keyPressed = jest.fn();

  beforeEach(() => {
    view = render(
      <SearchField
        query=""
        setQuery={setState}
        keyPressed={keyPressed}
        placeholder="Search test.."
      />
    );
  });

  it("should render as expected", () => {
    expect(view).toMatchSnapshot();
  });

  it("should render reset Button if non-empty query", () => {
    view.rerender(
      <SearchField
        query="search"
        setQuery={setState}
        keyPressed={keyPressed}
        placeholder="Search test.."
      />
    );

    const button: HTMLElement | null = screen.queryByRole("button");
    expect(button).toBeInTheDocument();
  });
});
