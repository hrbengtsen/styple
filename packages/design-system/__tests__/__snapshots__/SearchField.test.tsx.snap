// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SearchField should render as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="c-lesPJm c-lesPJm-icmpvrW-css"
      >
        <div
          class="c-lesPJm c-lesPJm-ibSoECS-css"
        >
          <svg
            class="lucide lucide-search"
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <line
              x1="21"
              x2="16.65"
              y1="21"
              y2="16.65"
            />
          </svg>
        </div>
        <input
          class="c-PdLBs c-PdLBs-bJUIkY-size-sm c-PdLBs-iewdWQl-css"
          placeholder="Search test.."
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="c-lesPJm c-lesPJm-icmpvrW-css"
    >
      <div
        class="c-lesPJm c-lesPJm-ibSoECS-css"
      >
        <svg
          class="lucide lucide-search"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <line
            x1="21"
            x2="16.65"
            y1="21"
            y2="16.65"
          />
        </svg>
      </div>
      <input
        class="c-PdLBs c-PdLBs-bJUIkY-size-sm c-PdLBs-iewdWQl-css"
        placeholder="Search test.."
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
